src/spec_concordius.cpp

```diff
@@ -0,0 +1,85 @@
+/************************************************************************/
+/*  Xania (M)ulti(U)ser(D)ungeon server source code                     */
+/*  (C) 2020 Xania Development Team                                     */
+/*  See the header to file: merc.h for original code copyrights         */
+/************************************************************************/
+
+#include "CHAR_DATA.hpp"
+#include "DescriptorFilter.hpp"
+#include "DescriptorList.hpp"
+#include "comm.hpp"
+#include "merc.h"
+#include "string_utils.hpp"
+#include <fmt/format.h>
+#include <string_view>
+#include <vector>
+
+using namespace fmt::literals;
+namespace {
+
+static inline constexpr auto SOCIAL_DICE = 30u;
+static inline constexpr auto NUM_SOCIAL_ACTIONS = 7u;
+
+inline static const std::vector<std::string> emotes{
+    "$n polishes $s golden claw.",
+    "$n looks skyward and closes $s eyes in thought.",
+    "$n raises $s hand and begins to whistle birdsong.",
+    "$n says 'I used to be an adventurer like you. But then Aquila talked some sense into me.'",
+    "$n sits down and begins to sharpen the edge of his shield against his heel.",
+    "$n says 'Beware the denizens of Yggdrasil to the west!'",
+    "$n tightens $s toga.",
+    "$n says 'How I long to visit Ultima once again...",
+    "$n says 'I'm told it's never sunny in Underdark. I don't like the sound of that.'",
+    "$n says 'Did you know there are over sixty different species of eagle?!'"};
+
+bool concordius_stands_guard(CHAR_DATA *ch) {
+
+    uint random = number_range(0, 100);
+    if (random > 90) {
+        act(emotes[random % emotes.size()], ch, nullptr, nullptr, To::Room);
+        return true;
+    }
+    bool found_someone = false;
+    for (auto &victim : descriptors().playing() | DescriptorFilter::same_room(*ch) | DescriptorFilter::to_character()) {
```

Comment:
(Aside: one can loop over `ch->in_room_people` too; though includes npcs too. I'm still developing an understanding of when one iteration is better than another)

Reply:
Yeah I had thought about that, but was trying to use the fancy new stuff. But I think I will swap to in_room as it is slightly more efficient. Just more old fashioned, that is until we make it a rangeable thing.